### ISSUE TITLE
Code style & columns naming in database

### ISSUE THREAD

**Issue by a-komarev** [yiisoft/yii2#6760]
In code style of Yii2 i found:
>Property names MUST be declared in camelCase.

But most part of components & modules using in database lowercase tables & columns names. *(I'm finding it more compatible too)*.
That means that we've got properties in ActiveRecord models like: `$this->mime_type`.
How to deal with it?
Is there any naming conventions for Yii2 application's databases? I understand that they couldn't be 100% strict because a lot of applications already got their schemas design, but it will be great if recomendations will be described in wiki.

Pretty good [description of naming conventions](https://launchbylunch.com/posts/2014/Feb/16/sql-naming-conventions/#naming-conventions).

**Comment by cebe** [comment]
You are free to choose the database naming style, yii does not have any restrictions on that except that the name must be a valid php identifier. If you do not want underscores in your php code you have to adjust your db schema to use camelCase, there is no conversion of attribute  names and we are not going to implement that.

**Comment by cebe** [comment]
I have added a note to the guide about this.

**Comment by a-komarev** [comment]
Thanks.
I've just thought about adding a note to [Code style page](https://github.com/yiisoft/yii2/blob/228f810912ac8c89d708b630d344f42669743625/docs/internals/core-code-style.md#42-properties). But you've done it faster :}

**Comment by cebe** [comment]
I think AR docs are more appropriate place for this because it is not really about defining properties but more about properties/attributes in AR.

**Comment by a-komarev** [comment]
Yes, thats right.
I understand that all namings in database is a matter of taste or/and company standards. And there is not a big deal to have different schemas naming for different components even.
I found it useful to read and start writing code on Yii using Code style page description. My applications looks more solid with framework structure and they are easier maintainable by another developers, because it's a documented standard.
Then I've tried to find out similar document about DB design, but found only Yii1.x guidelines.
Hope it will make this great framework more standartized and easier to start to work with it.
It could be not strict rules, but recomendations.
